### ISSUE TITLE
3D-Point-Cloud-Data-Processing:Validation exception while running the create_labeling_job api?

### ISSUE THREAD

**Issue by Suri12990** [aws/amazon-sagemaker-examples#1936]
Hello,

I have been trying to run create_labeling_api for the last two days, but I could not able to do it. I tried with different options, but no success. Can I get any help with this problem?
Here is the code:

```
## Set up Create Labeling Request

labelAttributeName = job_name + "-ref"

if task_type == "3DPointCloudObjectDetection":
    labelAttributeName = job_name

print(job_name)

ground_truth_request = {
        "InputConfig" : {
          "DataSource": {
            "S3DataSource": {
              "ManifestS3Uri": '{}'.format(manifest_s3_uri_map[task_type]),
            }
          },
        },
        "OutputConfig" : {
          "S3OutputPath": f's3://{BUCKET}/{EXP_NAME}/output/',
        },
        "HumanTaskConfig" : human_task_config,
        "LabelingJobName": job_name,
        "RoleArn": role, 
        "LabelAttributeName": labelAttributeName,
        "LabelCategoryConfigS3Uri": label_category_file_s3_uri_map[task_type]
    }

print(json.dumps(ground_truth_request, indent=4, sort_keys=True))



```
Ouput of the above code:

```
Point-Cloud-OD
{
    "HumanTaskConfig": {
        "AnnotationConsolidationConfig": {
            "AnnotationConsolidationLambdaArn": "arn:aws:lambda:eu-central-1:326177286064:function:ACS-3DPointCloudObjectDetection"
        },
        "MaxConcurrentTaskCount": 200,
        "NumberOfHumanWorkersPerDataObject": 1,
        "PreHumanTaskLambdaArn": "arn:aws:lambda:eu-central-1:326177286064:function:PRE-3DPointCloudObjectDetection",
        "TaskAvailabilityLifetimeInSeconds": 18000,
        "TaskDescription": "point-cloud-labelling",
        "TaskKeywords": [
            "lidar",
            "pointcloud"
        ],
        "TaskTimeLimitInSeconds": 360,
        "TaskTitle": "Detection-labelling-for-point-cloud",
        "UiConfig": {
            "HumanTaskUiArn": "arn:aws:sagemaker:eu-central-1:394669845002:human-task-ui/3DPointCloudObjectDetection"
        },
        "WorkteamArn": "arn:aws:sagemaker:eu-central-1:326177286064:workteam/private-crowd/suri"
    },
    "InputConfig": {
        "DataSource": {
            "S3DataSource": {
                "ManifestS3Uri": "s3://sample-point-cloud/artifacts/gt-point-cloud-demos/manifests/SingleFrame-manifest.json"
            }
        }
    },
    "LabelAttributeName": "Point-Cloud-OD",
    "LabelCategoryConfigS3Uri": "s3://sample-point-cloud/artifacts/gt-point-cloud-demos/label-category-config/label-category.json",
    "LabelingJobName": "Point-Cloud-OD",
    "OutputConfig": {
        "S3OutputPath": "s3://sample-point-cloud/validation/output/"
    },
    "RoleArn": "arn:aws:iam::326177286064:role/service-role/AmazonSageMaker-ExecutionRole-20210115T142880"
}

sagemaker_client.create_labeling_job(**ground_truth_request)
```

Error:
```
ClientError                               Traceback (most recent call last)
<ipython-input-220-84378bff2ae5> in <module>
----> 1 sagemaker_client.create_labeling_job(**ground_truth_request)
      2 print(f'Labeling Job Name: {job_name}')

~/anaconda3/envs/python3/lib/python3.6/site-packages/botocore/client.py in _api_call(self, *args, **kwargs)
    355                     "%s() only accepts keyword arguments." % py_operation_name)
    356             # The "self" in this scope is referring to the BaseClient.
--> 357             return self._make_api_call(operation_name, kwargs)
    358 
    359         _api_call.__name__ = str(py_operation_name)

~/anaconda3/envs/python3/lib/python3.6/site-packages/botocore/client.py in _make_api_call(self, operation_name, api_params)
    674             error_code = parsed_response.get("Error", {}).get("Code")
    675             error_class = self.exceptions.from_code(error_code)
--> 676             raise error_class(parsed_response, operation_name)
    677         else:
    678             return parsed_response

ClientError: An error occurred (ValidationException) when calling the CreateLabelingJob operation: HumanTaskUiArn is not supported for the Custom task type. You must provide an S3 URI identifying the location of a worker task template (UiTemplateS3Uri) for this task type. Remove the HumanTaskUiArn from your request, provide an UiTemplateS3Uri, and try your request again.
```
```

```

I went through all the documentation of boto3 and aws for create labelling job api, Still I could not able to resolve it. I think my error something has to do with deployment in eu-central-1?
Thanks in advance

**Comment by TEChopra1000** [comment]
Hi @Suri12990,

Are you trying to use the "build in task type" [3D point cloud object detection](https://docs.aws.amazon.com/sagemaker/latest/dg/sms-point-cloud-object-detection.html)?

If so, for `HumanTaskUiArn`, it looks like you're using:

`arn:aws:sagemaker:eu-central-1:394669845002:human-task-ui/3DPointCloudObjectDetection`

However, the supported human task UI ARN for 3D point cloud object detection (in eu-central-1) is: 

`arn:aws:sagemaker:eu-central-1:394669845002:human-task-ui/PointCloudObjectDetection`

Here's a reference: 
https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_UiConfig.html#sagemaker-Type-UiConfig-HumanTaskUiArn

Additionally, for pre- (PRE-) and post-processing (ACS-) lambda functions (`PreHumanTaskLambdaArn` and `AnnotationConsolidationLambdaArn` respectively) , you'll need to use the following ARNs (for eu-central-1):
* `arn:aws:lambda:eu-central-1:203001061592:function:PRE-3DPointCloudObjectDetection`
* `arn:aws:lambda:eu-central-1:203001061592:function:ACS-3DPointCloudObjectDetection`

References: 
* Pre-processing lambda: https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_HumanTaskConfig.html#sagemaker-Type-HumanTaskConfig-PreHumanTaskLambdaArn 
* Post-processing lambda: https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_AnnotationConsolidationConfig.html#sagemaker-Type-AnnotationConsolidationConfig-AnnotationConsolidationLambdaArn

**Comment by TEChopra1000** [comment]
@Suri12990 I'm going to close the ticket. Please reopen if you have any outstanding questions.

**Comment by Suri12990** [comment]
Hey @TEChopra1000 
thank you for the quick response. I made all necessary changes, but labeling job is not created, it has been showing in-progress for the last two days. I thought it take some time, but it is not created yet. 

Here is the output of the labeling job:
`{'LabelingJobStatus': 'InProgress', 'LabelCounters': {'TotalLabeled': 0, 'HumanLabeled': 0, 'MachineLabeled': 0, 'FailedNonRetryableError': 0, 'Unlabeled': 0}, 'CreationTime': datetime.datetime(2021, 1, 22, 13, 16, 57, 501000, tzinfo=tzlocal()), 'LastModifiedTime': datetime.datetime(2021, 1, 22, 13, 16, 58, 754000, tzinfo=tzlocal()), 'JobReferenceCode': '507354440914YHSw1jUyO0XYOgL2', 'LabelingJobName': 'point-label-1', 'LabelingJobArn': 'arn:aws:sagemaker:eu-central-1:507354440914:labeling-job/point-label-1', 'LabelAttributeName': 'point-label-1', 'InputConfig': {'DataSource': {'S3DataSource': {'ManifestS3Uri': 's3://point-cloud-data-1/artifacts/gt-point-cloud-demos/manifests/SingleFrame-manifest.json'}}, 'DataAttributes': {'ContentClassifiers': ['FreeOfPersonallyIdentifiableInformation', 'FreeOfAdultContent']}}, 'OutputConfig': {'S3OutputPath': 's3://point-cloud-data-1/examples/output/', 'KmsKeyId': ''}, 'RoleArn': 'arn:aws:iam::507354440914:role/service-role/AmazonSageMaker-ExecutionRole-20210122T130644', 'LabelCategoryConfigS3Uri': 's3://point-cloud-data-1/artifacts/gt-point-cloud-demos/label-category-config/label-category.json', 'StoppingConditions': {'MaxPercentageOfInputDatasetLabeled': 100}, 'HumanTaskConfig': {'WorkteamArn': 'arn:aws:sagemaker:eu-central-1:507354440914:workteam/private-crowd/daimer', 'UiConfig': {'HumanTaskUiArn': 'arn:aws:sagemaker:eu-central-1:394669845002:human-task-ui/PointCloudObjectDetection'}, 'PreHumanTaskLambdaArn': 'arn:aws:lambda:eu-central-1:203001061592:function:PRE-3DPointCloudObjectDetection', 'TaskKeywords': ['lidar', 'pointcloud'], 'TaskTitle': 'Add a Task Title Here - This is Displayed to Workers', 'TaskDescription': 'add a task description here', 'NumberOfHumanWorkersPerDataObject': 1, 'TaskTimeLimitInSeconds': 3600, 'TaskAvailabilityLifetimeInSeconds': 18000, 'MaxConcurrentTaskCount': 200, 'AnnotationConsolidationConfig': {'AnnotationConsolidationLambdaArn': 'arn:aws:lambda:eu-central-1:203001061592:function:ACS-3DPointCloudObjectDetection'}}, 'ResponseMetadata': {'RequestId': 'e1d35bf3-29ad-4a09-89aa-1e2fa3912cb1', 'HTTPStatusCode': 200, 'HTTPHeaders': {'x-amzn-requestid': 'e1d35bf3-29ad-4a09-89aa-1e2fa3912cb1', 'content-type': 'application/x-amz-json-1.1', 'content-length': '1848', 'date': 'Fri, 22 Jan 2021 13:16:59 GMT'}, 'RetryAttempts': 0}}
`

It is not showing any error this time, I successfully ran the script and also got the labeling job name.  
![aws_error](https://user-images.githubusercontent.com/53983084/105521375-17fdc680-5cdc-11eb-8bce-aee1ad047ecf.png)
